src/curve/stable/StableMintBurn.sol

```diff
@@ -110,8 +120,8 @@ contract StableMintBurn is StablePair {
             lTotalSupply = totalSupply;
         }
 
-        rAmount0 = liquidity * lReserve0 / lTotalSupply;
-        rAmount1 = liquidity * lReserve1 / lTotalSupply;
+        rAmount0 = liquidity.mulDiv(lReserve0, lTotalSupply);
+        rAmount1 = liquidity.mulDiv(lReserve1, lTotalSupply);
```

Comment:
decided to fix this as there is no guarantee that liquidity (on the same scale as `totalSupply`) will fit within 256-104=152 bits. Also recorded the reasoning in the spreadsheet